custom_components/plugwise/climate.py

```diff
@@ -234,13 +231,9 @@ def hvac_action(self) -> HVACAction:  # pw-beta add to Core
         self._previous_action_mode(self.coordinator)
 
         # Adam provides the hvac_action for each thermostat
-        if (control_state := self.device.get(CONTROL_STATE)) == HVACAction.COOLING:
-            return HVACAction.COOLING
-        if control_state == HVACAction.HEATING:
-            return HVACAction.HEATING
-        if control_state == HVACAction.PREHEATING:
-            return HVACAction.PREHEATING
-        if control_state == STATE_OFF:
+        if (control_state := self.device.get(CONTROL_STATE)) in (HVACAction.COOLING, HVACAction.HEATING, HVACAction.PREHEATING):
+            return control_state
+        if control_state == HVACMode.OFF:
```

Comment:
Does this work with mypy?

Reply:
The strange thing is some `.get` constructs do work in Core but not in beta ...

Reply:
I meant the `control_state` - `str` vs `HVACMode.Off` - `StrEnum`.